### PR TITLE
Prepare for 0.3.0 release.

### PR DESCRIPTION
**Pull request by dolaameng**
New release to enable Keras metrics integration.
Also fixed hardcoded version in test code.

FIX=b/361112861

### CODE REVIEW

File: tests/test_kaggle_api_client.py [hunk 1]
```diff
@@ -69,13 +70,13 @@ def test_download_corrupted_file_fail_integrity_check(self) -> None:
 
     @patch.dict("os.environ", {})
     def test_get_user_agent(self) -> None:
-        self.assertEqual(clients.get_user_agent(), "kagglehub/0.2.9")
+        self.assertEqual(clients.get_user_agent(), f"kagglehub/{kagglehub.__version__}")
```
Comment (stevemessick) [inline tests/test_kaggle_api_client.py:73]: Nice! I'll change `kaggle-api` to use that, too.

Comment (dolaameng) [reply tests/test_kaggle_api_client.py:73]: Thank you! Let me know if there is any other changes needed on the kagglehub side.

**Review by jeward414 — APPROVED** [review]
LGTM. Thanks for adding `kagglehub.__version__` to the tests!

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 ## Next (Unreleased)
 
+## v.0.3.0 (September 4th, 2024)
```
Comment (rosbo) [inline CHANGELOG.md:5]: Good call on the minor version bump given the Python support changes! TY

**Pull request merged**